pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -616,7 +660,7 @@ func (cb *ClusterBuilder) setUpstreamProtocol(cluster *clusterWrapper, port *mod
 	// h2. Clients would then connect with h2, while the upstream may not support it. This is not a
 	// concern for plaintext, but we do not have a way to distinguish https vs http here. If users of
 	// gateway want this behavior, they can configure UseClientProtocol explicitly.
-	if cb.sidecarProxy() && port.Protocol.IsUnsupported() {
+	if cb.sidecarProxy() && isAutoProtocol {
```

Comment:
What's the reason for this change?

Reply:
Just to check up the code as `isAutoProtocol` was created on L625